File: c/acronym/acronym.c
#include "acronym.h"

char *abbreviate(const char *phrase) {
  if (phrase == NULL)
    return NULL;
  if (strlen(phrase) == 0)
    return NULL;
  char buff[strlen(phrase)];
  int count = 0;
  int first = 1;

  for (int i = 0; phrase[i] != '\0'; i++) {
    if (isalpha(phrase[i]) && first == 1) {
      buff[count] = toupper(phrase[i]);
      count++;
      first = 0;
    }
    if (isspace(phrase[i]) || phrase[i] == '-') {
      first = 1;
    }
  }
  char *r = malloc(count * sizeof(char));
  strncpy(r, buff, count);
  r[count] = '\0';
  return r;
}

File: csharp/acronym/Acronym.cs
using System.Text.RegularExpressions;

public static class Acronym
{
    public static string Abbreviate(string phrase)
    {
        phrase = phrase.Replace("-", " ");
        phrase = Regex.Replace(phrase, "[^a-zA-Z ]*", "", RegexOptions.Compiled);
        var split = Regex.Split(phrase, " {1,}");


        List<char> acronym = new List<char>();

        foreach (string word in split)
        {
            acronym.Add(Char.ToUpper(word[0]));
        }
        string r = new string(acronym.ToArray());
        return r;
    }
}
